jupyter_server/extension/application.py

```diff
@@ -282,11 +282,13 @@ def initialize(self, serverapp, argv=[]):
         super(ExtensionApp, self).initialize(argv=argv)
         self.serverapp = serverapp
 
-        # Initialize config, settings, templates, and handlers.
-        self._prepare_config()
-        self._prepare_templates()
-        self._prepare_settings()
-        self._prepare_handlers()
+        # Initialize config, settings, templates, and handlers if serverapp has a web_app.
+        # Subcommands may not have a web_app attribute.
+        if hasattr(self.serverapp, 'web_app'):
```

Comment:
Hm, this doesn't seem right. I don't think we should be checking the serverapp for webapp initialization? It seems that we should make the ExtensionApp handle subcommands directly, rather than passing them to the ServerApp... I'll provide more thoughts below.